gateway/src/main.rs

```diff
@@ -38,6 +38,10 @@ async fn main() {
 
     let router = Router::new()
         .route("/inference", post(endpoints::inference::inference_handler))
+        .route(
+            "/batch_inference",
+            post(endpoints::batch_inference::prepare_batch_inference_handler),
```

Comment:
You mix `start_batch_inference` `prepare_batch_inference` `post_batch_inference` throughout.